backends/s3/s3.go

```diff
@@ -200,49 +200,74 @@ func (b *Backend) doList(ctx context.Context, prefix string) (simpleblob.BlobLis
 // Load retrieves the content of the object identified by name from S3 Bucket
 // configured in b.
 func (b *Backend) Load(ctx context.Context, name string) ([]byte, error) {
-	// Prepend global prefix
-	name = b.prependGlobalPrefix(name)
-
 	metricCalls.WithLabelValues("load").Inc()
 	metricLastCallTimestamp.WithLabelValues("load").SetToCurrentTime()
 
+	r, err := b.doLoadReader(ctx, name)
+	if err != nil {
+		metricCallErrors.WithLabelValues("load").Inc()
+		return nil, err
+	}
+	defer r.Close()
+
+	p, err := io.ReadAll(r)
+	if err = convertMinioError(err); err != nil {
+		metricCallErrors.WithLabelValues("load").Inc()
+		return nil, err
+	}
+	return p, nil
+}
+
+func (b *Backend) doLoadReader(ctx context.Context, name string) (io.ReadCloser, error) {
+	name = b.prependGlobalPrefix(name)
+
 	obj, err := b.client.GetObject(ctx, b.opt.Bucket, name, minio.GetObjectOptions{})
 	if err = convertMinioError(err); err != nil {
 		return nil, err
-	} else if obj == nil {
+	}
+	if obj == nil {
 		return nil, os.ErrNotExist
 	}
-
-	p, err := io.ReadAll(obj)
+	info, err := obj.Stat()
 	if err = convertMinioError(err); err != nil {
 		return nil, err
 	}
-	return p, nil
+	if info.Key == "" {
+		// minio will return an object with empty fields when name
+		// is not present in bucket.
+		return nil, os.ErrNotExist
+	}
+	return obj, nil
 }
 
 // Store sets the content of the object identified by name to the content
 // of data, in the S3 Bucket configured in b.
 func (b *Backend) Store(ctx context.Context, name string, data []byte) error {
-	// Prepend global prefix
-	name = b.prependGlobalPrefix(name)
+	metricCalls.WithLabelValues("store").Inc()
+	metricLastCallTimestamp.WithLabelValues("store").SetToCurrentTime()
 
 	info, err := b.doStore(ctx, name, data)
 	if err != nil {
+		metricCallErrors.WithLabelValues("store").Inc()
 		return err
 	}
 	return b.setMarker(ctx, name, info.ETag, false)
 }
 
 func (b *Backend) doStore(ctx context.Context, name string, data []byte) (minio.UploadInfo, error) {
-	metricCalls.WithLabelValues("store").Inc()
-	metricLastCallTimestamp.WithLabelValues("store").SetToCurrentTime()
```

Comment:
With this change we no longer count marker stores, while AWS will still bill you for them.

Reply:
Indeed, I'm moving it to `doStoreReader`